apps/coscrad-frontend/src/app/app.tsx

```diff
@@ -8,11 +9,37 @@ import { ResourceInfoContainer } from '../components/resource-info/resource-info
 import { TagDetailContainer } from '../components/tags/tag-detail.container';
 import { TagIndexContainer } from '../components/tags/tag-index.container';
 import { CategoryTreeContainer } from '../components/tree-of-knowledge/category-tree.container';
+import { getConfig } from '../config';
+import { fetchFreshState } from '../store/slices/utils/fetch-fresh-state';
 import './app.css';
+import { useAppDispatch } from './hooks';
 import { IndexToDetailFlowRoutes } from './index-to-detail-flow-routes';
 import { routes } from './routes/routes';
 
 export function App() {
+    const dispatch = useAppDispatch();
+
+    const eventSource = new EventSource(`${getConfig().apiUrl}/commands/notifications`);
+
+    eventSource.onmessage = (result) => {
+        /**
+         * TODO Move the following somewhere else. `store`?
+         *
+         * TODO Find a pattern that is closed to modification.
+         *
+         * TODO In the meantime, write a test that reminds us to update this
+         * with each new resource type.
+         */
+        const message = JSON.parse(result.data);
+
+        console.log('Incoming Message: ', JSON.parse(result.data));
+
+        const aggregateTypeFromMessage = message.aggregateCompositeIdentifier?.type;
+
+        if (isAggregateType(aggregateTypeFromMessage))
+            fetchFreshState(dispatch, aggregateTypeFromMessage);
```

Comment:
So this is requesting the full set of records for a given aggregate?  Could we just request the record by aggregate ID?  Or would this ignore other (external) state changes?  How does this message system work with other commands?  Like creating a connection between resources?

Reply:
I think you figured this out later. It is just a higher level abstraction that will defer to the appropriate fetch thunk for the aggregate type. Also, we can optimize later to do a `fetchById`, which would require a different fetch thunk and some logic to merge the update in Redux.